acis_thermal_check/state_builder.py

```diff
@@ -0,0 +1,294 @@
+import os
+import Ska.DBI
+from Chandra.Time import DateTime
+from pprint import pformat
+import Chandra.cmd_states as cmd_states
+import numpy as np
+from acis_thermal_check.utils import globfile
+
+class StateBuilder(object):
+    """
+    This is the base class for all StateBuilder objects. It
+    should not be used by itself, but subclassed.
+
+    Parameters
+    ----------
+    thermal_check : :class:`~acis_thermal_check.main.ACISThermalCheck` object
+        The ACISThermalCheck object this StateBuilder is attached to.
+    """
+    def __init__(self, thermal_check):
+        self.thermal_check = thermal_check
+        self.logger = self.thermal_check.logger
+
+    def set_options(self, args):
+        """
+        Give the StateBuilder arguments that were passed in
+        from the command line
+
+        Subclasses may want to overload this to do other things
+        with the arguments. 
+
+        Parameters
+        ----------
+        args : ArgumentParser object
+            Command line arguments
+        """
+        self.args = args
+
+    def get_prediction_states(self, tlm):
+        """
+        Get the states used for the thermal prediction.
+
+        Parameters
+        ----------
+        tlm : dictionary
+            Dictionary containg temperature and other telemetry
+        """
+        raise NotImplementedError("'StateBuilder should be subclassed!")
+
+    def get_validation_states(self, datestart, datestop):
+        """
+        Get states for validation of the thermal model.
+
+        Parameters
+        ----------
+        datestart : string
+            The start date to grab states afterward.
+        datestop : string
+            The end date to grab states before.
+        """
+        raise NotImplementedError("'StateBuilder should be subclassed!")
+
+class SQLStateBuilder(StateBuilder):
+    """
+    The SQLStateBuilder contains the original code used to 
+    obtain commanded states for prediction and validation of
+    a thermal model for a particular command load. It can also
+    be used for validation only.
+    """
+    def set_options(self, args):
```

Comment:
Could this be turned into an explicit set of keyword args instead of `args`?  Or at least a `**kwargs` dict with documented allowed options?  That would make the API and dependencies more clear, and make this module less tied to being used only via the ACIS thermal check tools.

Reply:
Yeah, this makes sense--should be an easy change.